great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -1019,6 +1105,23 @@ def run_profiler_on_data(
         )
         return result
 
+    def add_rule(self, rule: Rule) -> None:
+        """
+        Add Rule object to existing profiler object by appending to self._rules. In cases where a Rule with the same name exists, it is updated.
+        """
+        if not isinstance(rule, Rule):
+            raise TypeError("add_rule method requires a Rule to be passed in.")
+
+        index_to_delete: Optional[int] = None
+        for index in range(len(self._rules)):
+            existing_rule: Rule = self.rules[index]
+            if existing_rule.name == rule.name:
+                index_to_delete = index
```

Comment:
@Shinnnyshinshin This should not be needed -- there is a reconciliation strategy that would accomplish the same -- and using a high-level methods.  Might you be able to look into them?  Happy to help, if needed.  Thanks.

Reply:
Thank you again for this. The update to `add_rule()` has resulted in a true reconciliation of the Rule config, and update to both the `.rule` field  and `._profiler_config`